src/client/streamableHttp.test.ts

```diff
@@ -498,43 +512,6 @@ describe('StreamableHTTPClientTransport', () => {
 
         await transport['_startOrAuthSse']({});
         expect((actualReqInit.headers as Headers).get('x-custom-header')).toBe('CustomValue');
-
-        requestInit.headers['X-Custom-Header'] = 'SecondCustomValue';
```

Comment:
Why did we delete this test coverage?

I'm not sure the new test covers this scenario - this test seems to cover making changes to headers (`CustomValue` -> `SecondCustomValue` and that they get correctly updated).

Was this test not passing with the change? If not, can we just restore this?